src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1326,6 +1328,59 @@ else if (exercise instanceof TextExercise) {
                 .collect(Collectors.toList());
     }
 
+    public void testExportCourse_cannotCreateTmpDir() throws Exception {
+        Course course = database.createCourseWithTestModelingAndFileUploadExercisesAndSubmissions();
+        List<String> exportErrors = new ArrayList<>();
+
+        MockedStatic<Files> mockedFiles = mockStatic(Files.class);
+        mockedFiles.when(() -> Files.createDirectories(argThat(path -> path.toString().contains("exports")))).thenThrow(IOException.class);
+        Optional<Path> exportedCourse = courseExamExportService.exportCourse(course, courseArchivesDirPath, exportErrors);
+        mockedFiles.close();
+
+        assertThat(exportedCourse).isEmpty();
+    }
+
+    public void testExportCourse_cannotCreateCourseExercisesDir() throws Exception {
+        Course course = database.createCourseWithTestModelingAndFileUploadExercisesAndSubmissions();
+        List<String> exportErrors = new ArrayList<>();
+
+        MockedStatic<Files> mockedFiles = mockStatic(Files.class);
+        mockedFiles.when(() -> Files.createDirectory(argThat(path -> path.toString().contains("course-exercises")))).thenThrow(IOException.class);
+        Optional<Path> exportedCourse = courseExamExportService.exportCourse(course, courseArchivesDirPath, exportErrors);
+        mockedFiles.close();
+
+        assertThat(exportedCourse).isEmpty();
+    }
+
+    public void testExportCourseExam_cannotCreateTmpDir() throws Exception {
+        Course course = database.createCourseWithExamAndExercises();
+        List<String> exportErrors = new ArrayList<>();
+
+        Optional<Exam> exam = examRepo.findByCourseId(course.getId()).stream().findFirst();
+        assertThat(exam).isPresent();
+
+        MockedStatic<Files> mockedFiles = mockStatic(Files.class);
+        mockedFiles.when(() -> Files.createDirectories(argThat(path -> path.toString().contains("exports")))).thenThrow(IOException.class);
+        Optional<Path> exportedCourse = courseExamExportService.exportExam(exam.get(), courseArchivesDirPath, exportErrors);
+        mockedFiles.close();
+
+        assertThat(exportedCourse).isEmpty();
+    }
+
+    public void testExportCourseExam_cannotCreateExamsDir() throws Exception {
+        Course course = database.createCourseWithExamAndExercises();
+        List<String> exportErrors = new ArrayList<>();
```

Comment:
Wouldn't it make sense to also check the contents or count of this list in this test case and the ones above?